src/NUnitEngine/nunit.engine/RuntimeFramework.cs

```diff
@@ -53,6 +53,19 @@ public enum RuntimeType
     }
 
     /// <summary>
+    /// Enumeration identifying profile of .NET Framework
+    /// </summary>
+    public enum Profile
+    {
+        /// <summary>Profile Unspecified</summary>
+        Unspecified,
+        /// <summary>Client Profile</summary>
+        Client,
+        /// <summary>Full Profile</summary>
+        Full
+    }
```

Comment:
Have looked into this a little more. MS behaviour appears to be to not specify full profiles, only the restricted ones, which might lend something to @CharliePoole's comment that unspecified should mean full.
http://stackoverflow.com/a/3899324/1768779

The only issue I can think of then, is that in other methods such as CurrentFramework which don't check for Client Profile, we could potentially incorrectly specify .NETv4Client as Full, whereas at the moment Unspecified makes it clear the check doesn't take place. I don't know how this is used downstream however, maybe that will affect how it's best specified. :-)